### ISSUE TITLE
Regression: ctypes.Structure reports error when classmethod is called

### ISSUE THREAD

**Issue by einarwar** [microsoft/pyright#5932]
**Describe the bug**
* Classes inheriting ctypes.Structure will with pyright=1.1.326 flag the following error: `error: Object of type "_CField[Any, Any, Any]" is not callable (reportGeneralTypeIssues)`
* When testing with pyright=1.1.325 it passes

**Code or Screenshots**

```python
import ctypes
from typing import ClassVar


class MyStruct(ctypes.Structure):
    _fields_ = [("a", ctypes.c_uint32)]


class MyClass:
    _struct: ClassVar = MyStruct

    @classmethod
    def something(cls, encoded: bytes):
        return cls._struct.from_buffer_copy(encoded)


MyStruct.from_buffer_copy(b"1234")

```
Both the call inside `do_something()` and the last line will raise the error.  

**VS Code extension or command-line**
Running pyright from the commandline

**Comment by erictraut** [comment]
This was caused by [a recent change](https://github.com/python/typeshed/commit/4009efa3499519293f3520f986c0faaa9ff53216) to the typeshed stubs for `ctypes`. If you think this change was in error, please file a bug with the typeshed maintainers.

**Comment by erictraut** [comment]
I (partially) misdiagnosed this problem. I was correct in stating that the regression occurred due to a change in typeshed, but the typeshed change uncovered a bug in pyright. I've addressed the problem, and the fix will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.330, which I just published. It will also be included in an upcoming release of pylance.
